test/response_test.rb

```diff
@@ -852,9 +852,237 @@ class RubySamlTest < Minitest::Test
+
+      assert_raises(OneLogin::RubySaml::ValidationError, error_msg) do
+        response = OneLogin::RubySaml::Response.new(signed_message_encrypted_unsigned_assertion)
+        response.send(:assertion_encrypted?)
+      end
+
+      assert_raises(OneLogin::RubySaml::ValidationError, error_msg) do
+        response2 = OneLogin::RubySaml::Response.new(signed_message_encrypted_unsigned_assertion, :settings => settings)
+        response2.send(:assertion_encrypted?)
+      end
+
+      settings.certificate = ruby_saml_cert_text
+      settings.private_key = ruby_saml_key_text
+      assert_raises(OneLogin::RubySaml::ValidationError, error_msg) do
+        response3 = OneLogin::RubySaml::Response.new(signed_message_encrypted_unsigned_assertion)
+        response3.settings
+        response3.send(:assertion_encrypted?)
+      end
+    end
+
+    it 'raise if an encrypted assertion is found and the sp private key is wrong' do
+      settings.certificate = ruby_saml_cert_text
+      wrong_private_key = ruby_saml_key_text.sub!('A', 'B')
+      settings.private_key = wrong_private_key
+
+      error_msg = "Neither PUB key nor PRIV key: nested asn1 error"
+      assert_raises(OpenSSL::PKey::RSAError, error_msg) do
+        response = OneLogin::RubySaml::Response.new(signed_message_encrypted_unsigned_assertion, :settings => settings)
+        response.send(:assertion_encrypted?)
+      end
+   end
+
+    it 'return true if an encrypted assertion is found and settings initialized with private_key' do
+      settings.certificate = ruby_saml_cert_text
+      settings.private_key = ruby_saml_key_text
+      response = OneLogin::RubySaml::Response.new(signed_message_encrypted_unsigned_assertion, :settings => settings)
+      assert response.send(:assertion_encrypted?)
+
+      response2 = OneLogin::RubySaml::Response.new(signed_message_encrypted_signed_assertion, :settings => settings)
+      assert response2.send(:assertion_encrypted?)
+
+      response3 = OneLogin::RubySaml::Response.new(unsigned_message_encrypted_signed_assertion, :settings => settings)
+      assert response3.send(:assertion_encrypted?)
+
+      response4 = OneLogin::RubySaml::Response.new(unsigned_message_encrypted_unsigned_assertion, :settings => settings)
+      assert response4.send(:assertion_encrypted?)
+    end
+  end
+
+  describe "retrieve nameID and attributes from encrypted assertion" do
+
+    before do
+      settings.idp_cert_fingerprint = 'EE:17:4E:FB:A8:81:71:12:0D:2A:78:43:BC:E7:0C:07:58:79:F4:F4'
+      settings.issuer = 'http://rubysaml.com:3000/saml/metadata'
+      settings.assertion_consumer_service_url = 'http://rubysaml.com:3000/saml/acs'
+      settings.certificate = ruby_saml_cert_text
+      settings.private_key = ruby_saml_key_text
+    end
+
+    it 'able when signed_message_encrypted_unsigned_assertion' do
```

Comment:
`is possible when signed_message....` that's what `it`s are for, they should read fine with `it` and the following sentence
probably `is possible` are the words you were looking for =P